akka-docs/src/main/paradox/scala/stream/stream-flows-and-basics.md

```diff
@@ -354,4 +354,45 @@ been signalled already – thus the ordering in the case of zipping is defined b
 
 If you find yourself in need of fine grained control over order of emitted elements in fan-in
 scenarios consider using `MergePreferred`, `MergePrioritized` or `GraphStage` – which gives you full control over how the
-merge is performed.
\ No newline at end of file
+merge is performed.
+
+# Actor Materializer Life-cycle
+
+An important aspect of working with streams and actors is understanding an `ActorMaterializer`'s life-cycle.
+The materializer is bound to the life-cycle of the `ActorRefFactory` it is created from, which in practice will
+be either an `ActorSystem` or `ActorContext` (when the materializer is created within an `Actor`).
+
+As a quick reminder, the usual way of creating an `ActorMaterializer` is to create it next to your `ActorSystem`,
+which likely is in a "main" class of your application:
+
+Scala
+:   @@snip [FlowDocSpec.scala]($code$/scala/docs/stream/FlowDocSpec.scala) { #materializer-from-system }
+
+Java
+:   @@snip [FlowDocTest.java]($code$/java/jdocs/stream/FlowDocTest.java) { #materializer-from-system }
+
+This case is simple and nothing suprising can happen here. We simply would run streams on this materializer,
+and if the system terminates, the materializer (and all of its streams) would be terminated as well. 
```

Comment:
Maybe explain that when the materializer is shutdown streams that are still running are abruptly terminated on materializer shutdown already up here (since it is a little bit different from the normal complete or fail/cancel starting at one of the stages)